backend/src/services/release.ts

```diff
@@ -295,6 +326,168 @@ export async function getReleaseBySemver(user: UserInterface, modelId: string, s
   return release
 }
 
+function getSemverQueryBounds(querySemver: string) {
+  const semverRangeStandardised = semver.validRange(querySemver, { includePrerelease: false })
+
+  if (!semverRangeStandardised) {
+    throw BadReq(`Semver range, '${querySemver}' is invalid `)
+  }
+
+  const [expressionA, expressionB] = semverRangeStandardised.split(' ')
+
+  let lowerSemver: string = '',
+    upperSemver: string = '',
+    lowerInclusivity = false,
+    upperInclusivity = false
+  if (expressionA.includes('>')) {
+    if (expressionA.includes('=')) {
+      lowerSemver = expressionA.replace('>=', '')
+      lowerInclusivity = true
+    } else {
+      lowerSemver = expressionA.replace('>', '')
+    }
+    //do check for expressionB
+    if (expressionB) {
+      if (expressionB.includes('=')) {
+        upperSemver = expressionB.replace('<=', '')
+        upperInclusivity = true
+      } else {
+        upperSemver = expressionB.replace('<', '')
+      }
+    }
+  } else {
+    if (expressionA.includes('=')) {
+      upperSemver = expressionA.replace('<=', '')
+      upperInclusivity = true
+    } else {
+      upperSemver = expressionA.replace('<', '')
+    }
+  }
+
+  let lowerSemverObj: { metadata?: string | undefined; major: number; minor: number; patch: number } = {
```

Comment:
In a situation where two things are the exact same type, we should avoid redefining types. This allows us to have a single source of truth. If we ever updated that type for any reason, we'd only have to update it in one place and TS would guide us through updating all variables of that type.

In this scenario we should be using the `SemverObject` type you've already created in `backend/src/models/Release.ts` for both `lowerSemverObj` and `upperSemverObj`.

Reply:
Removed as was causing bugs and didn't seem strictly necessary.